runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -453,7 +427,7 @@ end
 
 --- Set underline for given diagnostics
 ---
---- Deprecated in favor of |vim.diagnostic.set_underline()|
+---@deprecated Prefer |vim.diagnostic.set_underline()|
```

Comment:
helps a lot to move the functions to a separate `lsp/_deprecated.lua` file, if possible.  Avoids noise and avoids confused contributors.

This `lsp.diagnostic` module can load the `lsp._deprecated` and merge it with its own module, for back-compat. We do something similar in the test infra: https://github.com/neovim/neovim/blob/13748512f6d6dfb5895c2233d2e07480e00eb753/test/unit/helpers.lua#L865-L868

Reply:
I'll push back on this a little bit: there are some local functions we use in `lsp/diagnostic.lua` to do the translation between LSP style data structures and `vim.diagnostic` data structures. If we split up that module, we either have to 1) duplicate those functions in both modules or 2) expose those local functions as part of the public interface. Both options are icky and I don't think worth the benefit of having a dedicated `_deprecated.lua` module.

Instead, what if we just tuck all the deprecated functions down at the bottom and throw in a "Deprecated" header and fold markers.